packages/block-editor/src/components/rich-text/index.js

```diff
@@ -378,7 +375,21 @@ export class RichText extends Component {
 		}
 
 		this.recalculateBoundaryStyle();
-		this.onSelectionChange();
+
+		// We know for certain that of focus, the old selection is invalid. It
+		// will be recalculated on `selectionchange`.
+		const index = undefined;
+		const activeFormats = undefined;
+
+		this.record = {
+			...this.record,
+			start: index,
+			end: index,
+			activeFormats,
+		};
+		this.props.onSelectionChange( index, index );
+		this.setState( { activeFormats } );
+		this.updateSelection();
```

Comment:
I suspect that this rework may also stabilise the split e2e tests more. Cc @gziolo, if you'd like to check if this is true.